src/connector.js

```diff
@@ -915,18 +1030,16 @@ function updateQuerySummaryState( newState ) {
 		querySummaryElement.textContent = "";
 		if ( querySummaryState.total > 0 ) {
 			let numberOfResults = querySummaryState.total.toLocaleString( params.lang );
+			// Create the <h2> element
+			const hTwoAnchor = document.createElement("h2");
 			// Generate the text content
-			const querySummaryHTML = ( ( querySummaryState.query !== "" && !params.isAdvancedSearch ) ? querySummaryTemplateHTML : noQuerySummaryTemplateHTML )
-				.replace( '%[numberOfResults]', numberOfResults )
-				.replace( '%[query]', '<span class="sr-query"></span>' )
-				.replace( '%[queryDurationInSeconds]', querySummaryState.durationInSeconds.toLocaleString( params.lang ) );
-
-			querySummaryElement.innerHTML = querySummaryHTML;
-
-			const queryElement = querySummaryElement.querySelector( '.sr-query' );
-			if ( queryElement ){
-				queryElement.textContent = querySummaryState.query;
-			}
+			const querySummaryText = ((querySummaryState.query !== "" && !params.isAdvancedSearch) ? querySummaryTemplateHTML : noQuerySummaryTemplateHTML)
+				.replace('%[numberOfResults]', numberOfResults)
+				.replace('%[query]', querySummaryState.query)
+				.replace('%[queryDurationInSeconds]', querySummaryState.durationInSeconds.toLocaleString(params.lang));
+			hTwoAnchor.textContent = querySummaryText;
+			querySummaryElement.innerHTML = "";
+			querySummaryElement.appendChild(hTwoAnchor);
```

Comment:
I don't believe your PR should be affecting this section; maybe you didn't do a rebase on the upstream?